src/impl/core/dappletContext.ts

```diff
@@ -12,23 +13,19 @@ import { GithubDappletProvider } from '../providers/githubDappletProvider';
 // instantated once on Wallet start
 export class DappletContext {
     private _dappletProvider: DappletProvider;
+    private _featuresRegistry: FeaturesRegistry;
 
-    constructor(config: ContextConfig) {
-        switch (config.source) {
-            case "github":
-                this._dappletProvider = new GithubDappletProvider();                
-                break;
-        
-            default:
-                this._dappletProvider = new GithubDappletProvider();    
-                break;
-        }
+    constructor(config?: ContextConfig) {
+        if (!config) config = DEFAULT_CONFIG;
+
+        this._dappletProvider = config.provider;
```

Comment:
`DappletContext` should support multiple `dappetProviders` to overcome provider failures.